db/schema.rb

```diff
@@ -462,6 +462,7 @@
     t.string "threatmetrix_review_status"
     t.string "threatmetrix_risk_rating"
     t.string "threatmetrix_policy_score"
+    t.boolean "inherited_proofing_proofed", default: false, null: false, comment: "When the user is proofed via inherited proofing"
```

Comment:
Wondering if we need the suffix here, "proofing components inherited proofing proofed" reads as a lot of "proof". I'm not exactly sure where in the inherited flow this happens, but in the standard unsupervised flow, the components here as stamped onto the profile only at the very end of the flow when the user enters their password, so I'd wonder if it could be misleading for the field to be named in a way which implies that they're already "proofed" prior to reaching this point.

```suggestion
    t.boolean "inherited_proofing", default: false, null: false, comment: "When the user is proofed via inherited proofing"
```

Reply:
Do I recall there being some desire for inherited proofing to be reusable? If so, should we consider assigning the value of the field to be the vendor from which the identity was inherited? This would be similar to how other fields behave, e.g. `document_check` is either `acuant` or `lexis_nexis` or `usps` depending on the vendor used for document authentication.

Reply:
@aduth this was my suggestion as well, but it appears that this bool is only being used to identify users who have _made it through the IP process_, not necessarily through the whole proofing process; and this, irrespective of service provider. This will be used to filter analytics data to include these users. Setting the SP as opposed to a bool is redundant and unnecessary because SP can be gleaned from the service_provided data on the User model.

Reply:
I didn't mean to suggest using the service provider as the value, but rather the vendor which was used to fulfill the proofing component (i.e. "VA"), in case we wanted to be able to allow this to be extensible to support other sources in the future, and for consistency with other fields in the table.

>this bool is only being used to identify users who have made it through the IP process, not necessarily through the whole proofing process

But the purpose of the table itself is as a sort of "temporary staging area" values for what will be stamped onto the profile at the end of the verification process, including this field.

>This will be used to filter analytics data to include these users.

On this point, will it be an issue for user actions which happen earlier in the flow, if this proofing component is only saved at the very last step of the InheritedProofingFlow ?

Reply:
> I didn't mean to suggest using the service provider as the value, but rather the vendor which was used to fulfill the proofing component (i.e. "VA"), in case we wanted to be able to allow this to be extensible to support other sources in the future, and for consistency with other fields in the table.

@aduth perhaps I am not following you. For IP, the service provider and the vendor are one and the same if I _am_ following you. IP pulls data from the SP which was previously "proofed" (to a greater or lesser degree) IOW with the proofing standards we use, what is it, IAL2?

All this field is signifying is whether or not the made it through the IP process and that's all we want and need to record. Any other "vendors" used for additional or more "strict" proofing will take place when IP hands the remaining proofing process to the IDV workflow.

> But the purpose of the table itself is as a sort of "temporary staging area" values for what will be stamped onto the profile at the end of the verification process, including this field.

Yes, we realize this.

> On this point, will it be an issue for user actions which happen earlier in the flow, if this proofing component is only saved at the very last step of the InheritedProofingFlow ?

No because this flag is only interested in recording whether or not the user made it through the IP proofing process, not whether or not this is an "IP user", or a user going through the IP process "in general". Remember, this flag is not interested whether or not the user makes it through the _whole_ proofing process, just _our_ IP proofing process. What takes place after IP hands off to IDV is of no consequence in relation to this flag.

Reply:
> For IP, the service provider and the vendor are one and the same if I _am_ following you.

I may defer to others (@jmhooper ?) who are more familiar with the requirements of inherited proofing, but given that a service provider is typically only loosely associated with an identity verified profile, I'd not expect to rely on it as a strong indicator of the elements of proofing, as the table is meant to support.

I could also imagine scenarios where a partner may have multiple discrete applications (separate issuer identifiers), but a shared proofing component source, and it would be simpler to reference a single vendor value which is consistent with other fields on the table, vs. a list of issuers per proofing source.

Could we assign it to the string result of [`inherited_proofing_service_provider`](https://github.com/18F/identity-idp/blob/346a7b38cc4817b3452d5e1c1fe241951a5e4f05/app/controllers/concerns/inherited_proofing_concern.rb#L13-L15)?

> No because this flag is only interested in recording whether or not the user made it through the IP proofing process, not whether or not this is an "IP user", or a user going through the IP process "in general". Remember, this flag is not interested whether or not the user makes it through the _whole_ proofing process, just _our_ IP proofing process. What takes place after IP hands off to IDV is of no consequence in relation to this flag.

Ok, I was mistakenly under the impression that there were other screens within the flow that weren't part of the `FlowStateMachine` implementation (specifically cancellation behavior), in which case it wouldn't have been reflected in those analytics. Unsure if there are others, but I see that cancellation at least was duplicated rather than extending the default IdV behavior so, while it wouldn't include the proofing components, I suppose we could still reference it by a combination of the event's `analytics_id` and `issuer`.

Reply:
@aduth 

>I'd not expect to rely on it as a strong indicator of the elements of proofing, as the table is meant to support.

It is _not_ supposed to be an indicator of whether or not the user has been "proofed" per se, it is simply to indicate the user has gone through the IP proofing _process_. The user isn't considered "proofed" until they go through the _whole_ process which includes the IP process and the _remaining_ IDV process. Whether or not this value is a good fit for the table in question is something else, but this indicator is simply about getting through the IP _process_. If the user traverses the IP process, this indicator simply indicates the user has gone _through_ the IP process.